java/java.hints.declarative/src/org/netbeans/modules/java/hints/declarative/conditionapi/Matcher.java

```diff
@@ -90,25 +91,38 @@ public Void scan(Tree tree, Void p) {
         return result[0];
     }
 
+    @SuppressWarnings("BoxedValueEquality")
     public boolean referencedIn(@NonNull Variable variable, @NonNull Variable in) {
-        final Element e = ctx.ctx.getInfo().getTrees().getElement(ctx.getSingleVariable(variable));
+        final Trees trees = ctx.ctx.getInfo().getTrees();
+        final Element e = trees.getElement(ctx.getSingleVariable(variable));
 
         if (e == null) { //TODO: check also error
             return false;
         }
 
+        TreePath v = ctx.getSingleVariable(in);
+        if (v != null && e.equals(trees.getElement(v))) {
```

Comment:
Good catch, but it would be better, I think, to move this test into the for loop (and operate over `tp`). We could have something like: `Arrays.asList(0, 1, b, 3)` with a pattern list `Arrays.asList($arrgs$)`.

Reply:
haven't considered that. Fixed + added another test for the mentioned case.